### ISSUE TITLE
async derive does not get awaited

### ISSUE THREAD

**Issue by lithdew** [elysiajs/elysia#588]
### What version of Elysia.JS is running?

1.0.11

### What platform is your computer?

Darwin 23.4.0 arm64 arm

### What steps can reproduce the bug?

```ts
export async function deriveSession({ cookie }: Context<any>) {
  return { session: "123" };
}
```

Use the `derive()` above as `.derive(deriveSession)`. The field `session` is undefined in all handlers.

**Comment by SaltyAom** [comment]
Should have been fixed in 1.0.14.

**Comment by SaltyAom** [comment]
Closing as complete, as unable to reproduce on the latest version of Elysia and lack of interaction.

Feels free to reopen the issue if the problem still persist.
